app/src/main/java/com/example/photogalleryapp/SearchActivity.java

```diff
@@ -33,25 +37,87 @@ protected void onCreate(Bundle savedInstanceState) {
+                        .get(Calendar.YEAR), myCalendar.get(Calendar.MONTH),
+                        myCalendar.get(Calendar.DAY_OF_MONTH)).show();
+            }
+        });
+
+        DatePickerDialog.OnDateSetListener endDatePicker = new DatePickerDialog.OnDateSetListener() {
+            @Override
+            public void onDateSet(DatePicker view, int year, int monthOfYear,
+                                  int dayOfMonth) {
+                myCalendar.set(Calendar.YEAR, year);
+                myCalendar.set(Calendar.MONTH, monthOfYear);
+                myCalendar.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+                updateDateLabel(endDate);
+            }
+        };
+
+        endDate.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                new DatePickerDialog(SearchActivity.this, endDatePicker, myCalendar
+                        .get(Calendar.YEAR), myCalendar.get(Calendar.MONTH),
+                        myCalendar.get(Calendar.DAY_OF_MONTH)).show();
+            }
+        });
+
 
-        startDate.setText(today);
-        endDate.setText(tomorrow);
     }
 
+    private void updateDateLabel(EditText etDate) {
+        String myFormat = "yyyyMMdd"; //In which you need put here
+        SimpleDateFormat sdf = new SimpleDateFormat(myFormat);
+        etDate.setText(sdf.format(myCalendar.getTime()));
+    }
+
+
     public void submitButton(View view) {
-        // create a new intent, retrieve the search data and
-        // return value to MainActivity (GalleryActivity)
+        boolean validation = validateSearchInput();
 
-        Intent intent = new Intent();
+        if (validation) {
+            // create a new intent, retrieve the search data and
+            // return value to MainActivity (GalleryActivity)
 
-        // Set intent extra data with value from the text input
-        intent.putExtra("startDate", startDate.getText().toString());
-        intent.putExtra("endDate", endDate.getText().toString());
-        intent.putExtra("editKeywordSearch", editKeywordSearch.getText().toString());
+            Intent intent = new Intent();
 
-        // Return ok signal
-        setResult(RESULT_OK, intent);
-        finish();
+            // Set intent extra data with value from the text input
+            intent.putExtra("startDate", startDate.getText().toString());
+            intent.putExtra("endDate", endDate.getText().toString());
+            intent.putExtra("editKeywordSearch", editKeywordSearch.getText().toString());
+            intent.putExtra("latitude", editLat.getText().toString());
+            intent.putExtra("longitude", editLong.getText().toString());
+
+            // Return ok signal
+            setResult(RESULT_OK, intent);
+            finish();
+        }
+    }
+
+    private boolean validateSearchInput() {
+        // TODO: Basic validation
```

Comment:
:eyes: if we decide to extend the photo gallery app as our group's project (past sprint 3) this would make a good feature update